File: dotnet/Samples/Translations/File/Batch/CloudBatchFileTranslationService.cs
using LanguageWeaver.Sdk.Common.Cloud;
using LanguageWeaver.Sdk.Constants;
using LanguageWeaver.Sdk.Translate.Cloud.Request;
using LanguageWeaver.Sdk.Translate.Cloud.Result;

namespace LanguageWeaver.Sdk.Samples.Translations.File.Batch;

public class CloudBatchFileTranslationService
{
    public static void Main()
    {
        using var lwClient = new CloudLanguageWeaverClient().Build();

        TranslateBatchFile(lwClient);
        TranslateBatchFileWithLinguisticOptions(lwClient);
    }

    private static void TranslateBatchFile(CloudLanguageWeaverClient lwClient)
    {
        var translateBatchFileRequest = new CloudTranslateBatchFileRequest
        {
            SourceLanguageId = "eng",
            TargetLanguageId = "fra",
            Model = "generic",
            PdfConverter = PdfConverter.STANDARD,
            Dictionaries = new List<string>
            {
                "689f06cf-36ba-4903-a530-da1f7766f478",
                "3d297ee3-0878-4ef7-9ee7-ca14b48e6956"
            },
            // provide full path to the input and output folders
            InputBatchFolderPath = Path.Combine(
                Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input"),
            OutputBatchFolderPath = Path.Combine(
                Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Output")
        };

        CloudTranslationBatchFileResult cloudTranslationBatchFileResult =
            lwClient.TranslateBatchFile(translateBatchFileRequest);
    }

    private static void TranslateBatchFileWithLinguisticOptions(CloudLanguageWeaverClient lwClient)
    {
        var translateBatchFileRequest = new CloudTranslateBatchFileRequest()
        {
            SourceLanguageId = "fra",
            TargetLanguageId = "eng",
            Model = "generic",
            LinguisticOptions = new Dictionary<string, string>
            {
                { "Spelling", "US" }
            },
            Dictionaries = new List<string>
            {
                "689f06cf-36ba-4903-a530-da1f7766f478",
                "3d297ee3-0878-4ef7-9ee7-ca14b48e6956"
            },
            // provide full path to the input and output folders
            InputBatchFolderPath = Path.Combine(
                Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input"),
            OutputBatchFolderPath = Path.Combine(
                Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Output")
        };

        CloudTranslationBatchFileResult cloudTranslationBatchFileResult =
            lwClient.TranslateBatchFile(translateBatchFileRequest);
    }
}

File: javascript/backend/samples/translations/file/batch/CloudBatchFileTranslationService.js
const path = require("path");
const {CloudLanguageWeaverClient, CloudTranslateBatchFileRequest, PdfConverter} = require("@language-weaver/lw-sdk-js");

const translateBatchFile = async () => {
    try {
        const cloudLanguageWeaverClient = await new CloudLanguageWeaverClient().build();
        const translateFileRequest = new CloudTranslateBatchFileRequest();
        translateFileRequest.sourceLanguageId = "eng";
        translateFileRequest.targetLanguageId = "fra";
        translateFileRequest.model = "generic";
        // provide list of dictionaries
        // translateFileRequest.addDictionary("689f06cf-36ba-4903-a530-da1f7766f478");
        // translateFileRequest.addDictionary("3d297ee3-0878-4ef7-9ee7-ca14b48e6956");
        // provide full path to the input and output folders
        translateFileRequest.input = path.resolve("resources/input");
        translateFileRequest.outputDir = path.resolve("resources/output");
        await cloudLanguageWeaverClient.translateBatchFileUsingCloudParams(translateFileRequest);
    } catch (e) {
        console.log(e);
    }
}

const translatePdfBatchFile = async () => {
    try {
        const cloudLanguageWeaverClient = await new CloudLanguageWeaverClient().build();
        const translateFileRequest = new CloudTranslateBatchFileRequest();
        translateFileRequest.sourceLanguageId = "eng";
        translateFileRequest.targetLanguageId = "fra";
        translateFileRequest.model = "generic";
        translateFileRequest.pdfConverter = PdfConverter.STANDARD;
        // provide full path to the input and output folders
        translateFileRequest.input = path.resolve("resources/input_pdf");
        translateFileRequest.outputDir = path.resolve("resources/output");
        await cloudLanguageWeaverClient.translateBatchFileUsingCloudParams(translateFileRequest);
    } catch (e) {
        console.log(e);
    }
}

const translateBatchFileWithLinguisticOptions = async () => {
    try {
        const cloudLanguageWeaverClient = await new CloudLanguageWeaverClient().build();
        const translateFileRequest = new CloudTranslateBatchFileRequest();
        translateFileRequest.sourceLanguageId = "fra";
        translateFileRequest.targetLanguageId = "eng";
        translateFileRequest.model = "generic";
        // provide full path to the input and output folders
        translateFileRequest.input = path.resolve("resources/input_linguistic_options");
        translateFileRequest.outputDir = path.resolve("resources/output");

        const linguisticOptions = new Map();
        linguisticOptions.set("Spelling", "UK");
        translateFileRequest.linguisticOptions = linguisticOptions;

        await cloudLanguageWeaverClient.translateBatchFileUsingCloudParams(translateFileRequest);
    } catch (e) {
        console.log(e);
    }
}

translateBatchFile();
// translatePdfBatchFile();
// translateBatchFileWithLinguisticOptions();